core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -422,6 +422,37 @@ private[spark] class MesosClusterScheduler(
     desc.schedulerProperties.get("spark.cores.max").map { v =>
       options ++= Seq("--total-executor-cores", v)
     }
+
+    /**
+      * The properties to pass along from Spark Submit to the Mesos Scheduler.
+      * http://spark.apache.org/docs/1.5.1/configuration.html#available-properties
+      */
+    val propertiesToPropagate = Set(
+      "spark.driver.extraClassPath",
```

Comment:
I see, the idea is that we pass all configuration to run the spark job as if you're running it locally to the scheduler to forward, so when it runs somewhere in the cluster it should have the same configurations. 

I think the right fix is to include everything and allow overrides like the ones you mentioned. I need to look into more if what other flags we need to consider, @andrewor14 do you any other flags we need to capture?